open-codegen/opengen/templates/optimizer.rs.template

```diff
@@ -77,29 +77,42 @@ const MAX_DURATION_MICROS: u64 = {{solver_config.max_duration_micros}};
 {% endif %}
 
 {% if activate_clib_generation -%}
-/// C version of ExitStatus
+/// Solver cache
+#[allow(non_camel_case_types)]
+pub struct {{meta.optimizer_name}}Cache {
+    cache: HomotopyCache,
+}
+
+impl {{meta.optimizer_name}}Cache {
+    pub fn new(cache: HomotopyCache) -> Self {
+        {{meta.optimizer_name}}Cache { cache }
+    }
+}
+
+/// {{meta.optimizer_name}} version of ExitStatus
+#[allow(non_camel_case_types)]
 #[repr(C)]
-pub enum CExitStatus {
+pub enum {{meta.optimizer_name}}ExitStatus {
     /// The algorithm has converged
     ///
     /// All termination criteria are satisfied and the algorithm
     /// converged within the available time and number of iterations
-    Converged,
+    {{meta.optimizer_name}}Converged,
```

Comment:
Do you think we can change `{{meta.optimizer_name}}Converged` to `Converged` (same for other items)? I guess it suffices to have `{{meta.optimizer_name}}ExitStatus`.

Reply:
No, C naming causes collisions (this works in C++ though)

Reply:
OK then - just a side note: `meta.optimizer_name` should be a string that contains no special characters other than underscores, does not contain white spaces and does not start with a number. It would be a good idea to add an assertion in Python.

Reply:
Indeed, I will add this check as well

Reply:
Added!